File: src/main/java/com/medicalsystem/model/ApplicationUser.java
package com.medicalsystem.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "APPLICATION_USERS")
public class ApplicationUser {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    @Getter @Setter
    private long id;

    @Column(name = "FULLNAME")
    @Getter @Setter
    private String fullname;

    @Column(name = "EMAIL")
    @Getter @Setter
    private String email;

    @Column(name = "USERNAME", unique = true)
    @Getter @Setter
    private String username;

    @Column(name = "PASSWORD")
    @Getter @Setter
    private String password;

    @Column(name = "STATUS")
    @Getter @Setter
    private String status;

    @Column(name = "ADMIN")
    @Getter @Setter
    private boolean admin;

}

File: src/main/kotlin/com/medicalsystem/model/ApplicationUser.kt
package com.medicalsystem.model

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import javax.persistence.*

@Entity
@Table(name = "USERS")
data class ApplicationUser(

        @Id @GeneratedValue
        @Column(name = "ID")
        var id: Long = 0,

        @Column(name = "FULL_NAME")
        var fullName: String = "",

        @Column(name = "EMAIL", unique = true)
        var email: String = "",

        @Column(name = "USERNAME", unique = true)
        var username: String = "",

        @Column(name = "PASSWORD")
        var password: String = "",

        @Column(name = "STATUS")
        var status: String = "",

        @Column(name = "ADMIN")
        var admin: Boolean = false
) {
    @JsonIgnore
    fun getAuthorities(): List<GrantedAuthority> =
            if (admin) {
                listOf(SimpleGrantedAuthority("user"), SimpleGrantedAuthority("admin"))
            } else {
                listOf(SimpleGrantedAuthority("user"))
            }
}